src/Authorization.AspNetCore/AuthorizationError.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Linq;
+using System.Text;
+using GraphQL.Language.AST;
+using GraphQL.Validation;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Infrastructure;
+
+namespace GraphQL.Server.Authorization.AspNetCore
+{
+    /// <summary>
+    /// An error that represents an authorization failure while parsing the document
+    /// </summary>
+    public class AuthorizationError : ValidationError
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorizationError"/> class for a specified authorization result
+        /// </summary>
+        public AuthorizationError(INode node, ValidationContext context, OperationType? operationType, AuthorizationResult result)
+            : this(node, context, GenerateMessage(operationType, result), result)
+        {
+            OperationType = operationType;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorizationError"/> class for a specified authorization result with a specific error message
+        /// </summary>
+        public AuthorizationError(INode node, ValidationContext context, string message, AuthorizationResult result)
+            : base(context.OriginalQuery, "6.1.1", message, node == null ? Array.Empty<INode>() : new INode[] { node })
+        {
+            Code = "authorization";
+            AuthorizationResult = result;
+        }
+
+        /// <summary>
+        /// Returns the result of the ASP.NET Core authorization request
+        /// </summary>
+        public virtual AuthorizationResult AuthorizationResult { get; }
+
+        /// <summary>
+        /// The GraphQL operation type
+        /// </summary>
+        public OperationType? OperationType { get; }
+
+        /// <summary>
+        /// Gets a <see cref="StringBuilder"/> containing the error message header for this <see cref="AuthorizationError"/>
+        /// </summary>
+        /// <returns></returns>
+        public StringBuilder GetErrorStringBuilder() => GetErrorStringBuilder(OperationType);
```

Comment:
I would not expose this method.

Reply:
Or at least change it to   `public static void AppendFailureHeader(StringBuilder error)` to align with `AppendFailureLine`.